backend/app/compat.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime
+from typing import Optional
+
+import requests
+from fastapi import APIRouter, FastAPI
+
+from . import apps, db, search
+
+router = APIRouter(prefix="/compat")
+
+
+def register_to_app(app: FastAPI):
+    app.include_router(router)
+
+
+def get_repo_creation_date(appid: str) -> Optional[str]:
+    if created_at := db.redis_conn.get(f"{appid}:added_at"):
+        return created_at
+    else:
+        try:
+            github_repo = requests.get(f"https://api.github.com/repos/flathub/{appid}")
+            github_repo.raise_for_status()
+            created_at = github_repo.json().get("created_at")
+            db.redis_conn.set(f"added_at:{appid}", created_at)
+            return created_at
+        except requests.exceptions.RequestException:
+            return None
+
+
+def get_release_date(app) -> Optional[str]:
+    if release_ts := app.get("releases")[0].get("timestamp"):
+        release_ts = int(release_ts)
+        return datetime.utcfromtimestamp(release_ts).strftime("%Y-%m-%d")
+
+    return None
+
+
+def get_app_blurb(key: str):
+    blurb = None
+    if app := db.get_json_key(key):
+        appid = key[5:]
+        blurb = {
+            "flatpakAppId": appid,
+            "name": app.get("name"),
+            "summary": app.get("summary"),
+            "currentReleaseVersion": app.get("releases")[0].get("version"),
+            "currentReleaseDate": get_release_date(app),
+            "iconDesktopUrl": app.get("icon"),
+            "iconMobileUrl": app.get("icon"),
+            "inStoreSinceDate": get_repo_creation_date(appid),
+        }
+
+    return blurb
+
+
+def list_apps_in_index(index="apps:index"):
+    appids = sorted(db.redis_conn.smembers(index))
+    ret = []
+
+    for appid in appids:
+        if app := get_app_blurb(appid):
+            ret.append(app)
+
+    return ret
+
+
+@router.get("/apps")
+def get_apps():
+    return list_apps_in_index()
+
+
+@router.get("/apps/category/{category}")
+def get_apps_in_category(category: str):
+    return list_apps_in_index(f"categories:{category}")
+
+
+@router.get("/apps/collection/recently-updated")
```

Comment:
Other collections will be needed too I guess?

Reply:
This is the only collection in the v1.

Reply:
I would have thought, that there is at least a `new` collection in use https://github.com/flathub/linux-store-backend/blob/da80c7bc96c9d7f6cb30723d7e1cf2e413a0a766/src/main/java/org/flathub/api/service/ApiServiceImpl.java#L21

Reply:
It's used only for the RSS feed: https://flathub.org/api/v1/apps/collection/new/feed

We already implement it in the v2: https://github.com/flathub/website/blob/main/backend/app/main.py#L205=